include/pressio/ops/eigen/ops_level2.hpp

```diff
@@ -83,19 +94,25 @@ product(::pressio::nontranspose /*unused*/,
 	const beta_t & beta,
 	y_type & y)
 {
-
   assert( ::pressio::ops::extent(y, 0) == ::pressio::ops::extent(A, 0) );
   assert( ::pressio::ops::extent(x, 0) == ::pressio::ops::extent(A, 1) );
 
+  using sc_t = typename ::pressio::Traits<y_type>::scalar_type;
+  constexpr sc_t zero{0};
+  const sc_t alpha_(alpha);
+  const sc_t beta_(beta);
+  const bool has_beta = beta_ != zero;
   auto & y_n = impl::get_native(y);
   const auto & A_n = impl::get_native(A);
   const auto & x_n = impl::get_native(x);
-  if (beta == static_cast<beta_t>(0)){
-    y_n = alpha * A_n * x_n;
-  }
-  else{
-    y_n = beta * y_n + alpha * A_n * x_n;
+  if (alpha_ == zero) {
+    if (has_beta) { ::pressio::ops::scale(y_n, beta_); }
+    else { ::pressio::ops::set_zero(y_n); }
```

Comment:
@fnrizzi Would you agree that we should put `set_zero` variant _inside_ `pressio::ops::scale()` to obtain general protection against NaN injection there ?

Reply:
yes i agree!